tests/test_auth.py

```diff
@@ -7,17 +9,45 @@
 class TestAuth(AsyncTestCase):
 
     def setUp(self):
-        self.auth_args = ('auth_url', 'username', 'password', 'mock_project_name',
-                          'mock_user_domain_name', 'mock_project_domain_name')
+        self.auth_args = ('http://url', 'm_user', 'm_pass', 'm_project',
+                          'm_user_domain', 'm_project_domain')
         self.auth = AuthPassword(*self.auth_args)
 
     def tearDown(self):
         patch.stopall()
+        for name in list(os.environ.keys()):
+            if name.startswith('OS_'):
+                del os.environ[name]
 
     async def test_create_object(self):
-        self.assertTrue(self.auth.auth_url.endswith("/auth/tokens"))
+        expected_payload = {'auth': {
+            'identity': {'methods': ['password'], 'password': {'user': {
+                'domain': {'name': 'm_user_domain'},
+                'name': 'm_user', 'password': 'm_pass'
+            }}},
+            'scope': {'project': {'domain': {'name': 'm_project_domain'}, 'name': 'm_project'}}
+        }}
+        self.assertEqual(self.auth._auth_payload, expected_payload)
+        self.assertEqual(self.auth._auth_endpoint, 'http://url/auth/tokens')
         self.assertTrue('Content-Type' in self.auth.headers)
 
+    async def test_create_object_use_environ(self):
+        expected_payload = {'auth': {
+            'identity': {'methods': ['password'], 'password': {'user': {'domain': {'name': 'udm'}, 'name': 'uuu', 'password': 'ppp'}}},
+            'scope': {'project': {'domain': {'name': 'udm'}, 'name': 'prj'}}
+        }}
+        env = {
+            'OS_AUTH_URL': 'https://keystone',
+            'OS_PASSWORD': 'ppp', 'OS_USERNAME': 'uuu',
+            'OS_USER_DOMAIN_NAME': 'udm', 'OS_PROJECT_NAME': 'prj'
+        }
+
+        with Environment(env):
```

Comment:
maybe it can help -> "with patch.dict('os.environ', clear=True, env):"
https://docs.python.org/3.4/library/unittest.mock.html#unittest.mock.patch.dict

Reply:
Even better, done